src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -189,44 +189,106 @@ namespace OpenLoco::World::MapGenerator
         return ((randVal & 0xFF) * landObj->numVariations) >> 8;
     }
 
+    static void applySurfaceStyleToMarkedTiles(HeightMap& heightMap, uint8_t surfaceStyle, bool requireMark)
+    {
+        for (auto pos : World::getDrawableTileRange())
+        {
+            const bool tileIsMarked = heightMap.isMarkerSet({ pos.x, pos.y });
+            if ((requireMark && !tileIsMarked) || (!requireMark && tileIsMarked))
```

Comment:
this is an XOR, which can be simplfied to `if (!requireMark != !tileIsMarked)`

Reply:
Nice idea. I suppose we could simplify it even further to `if (requireMark != tileIsMarked)`, right?

Reply:
oh yeah good point, i'm not sure why i needed the negations in there :|